File: FitnessApp-master/Services/MealsService.cs
using fitnessapp.Models.Food;
using fitnessData.AppData;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Linq;

namespace fitnessapp.Services
{
    public class MealsService
    {
        private AppDataDbContext _context;

        public MealsService(AppDataDbContext context)
        {
            _context = context;
        }

        public Meals_List List(int userid)
        {
            Meals_List model = new Meals_List();

            if (_context.Meals.Any(m => m.UserId == userid))
            {
                var meals = _context.Meals
                    .Include(m => m.Dishes)
                    .Where(m => m.UserId == userid)
                    .ToList();

                foreach (var meal in meals)
                {
                    var mealsList = new List<Dish>();

                    if (meal.Dishes.Count > 0)
                    {
                        foreach (var e in meal.Dishes)
                        {
                            var b = _context.Dishes.Find(e.DishId);
                            mealsList.Add(b);
                        }
                    }

                    model.Meals.Add(new MealVM(meal.Id, meal.DateTime, mealsList));
                }
            }

            return model;
        }

        public void Add(Meals_Create model)
        {
            var meal = new Meal(model.userid, model.datetime);

            List<MealDish> dishList = new List<MealDish>();
            var dish = _context.Dishes.Find(model.dishid);
            dishList.Add(new MealDish(meal, dish));

            meal.Dishes = dishList;

            _context.Add(meal);
            _context.SaveChanges();
        }

        public void Delete(int mealid)
        {
            var meal = _context.Meals.Find(mealid);

            _context.Remove(meal);
            _context.SaveChanges();
        }
    }
}

File: FitnessApp-master/ClientApp/src/Services/MealsService.js
import axios from 'axios';
import Dish from '../Models/Dish';
import Meal_List from '../Models/Meal_List';


export default class MealsService {

    static Dishes(id, callback) {

        axios.get('meals/dishes/', {
            params: {
                id: id
            }
        })
            .then(function (res) {
                if (res.data.length > 0) {

                    let dishlist = [];

                    for (let i = 0; i < res.data.length; i++) {
                        dishlist.push(
                            new Dish(
                                res.data[i].id,
                                res.data[i].name,
                                res.data[i].description)
                        )
                    };

                    callback(dishlist);
                }
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }

    static List(id, callback) {

        axios.get('meals/list/', {
            params: {
                id: id
            }
        })
            .then(function (res) {
                if (res.data.meals.length > 0) {
                    let mealsslist = [];

                    for (var i = 0; i < res.data.meals.length; i++) {
                        let meal = res.data.meals[i];
                        let dishes = meal.dishes;
                        let dishlist = [];

                        for (let i = 0; i < dishes.length; i++) {
                            dishlist.push(
                                new Dish(
                                    dishes[i].id,
                                    dishes[i].name,
                                    dishes[i].description)
                            )
                        };

                        let model = new Meal_List(
                            meal.id,
                            meal.datetime,
                            dishlist
                        );
                        mealsslist.push(model);
                    }
                    callback(mealsslist);
                }
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }

    static delete(id, userid, callback) {

        axios.delete('meals/delete/', { params: { id: id } })
            .then(function (res) {
                var response = res;
                MealsService.List(userid, callback);
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }

    static add(userid, meal, callback) {

        axios.post('meals/add/',
            {
                "userid": userid,
                "datetime": meal.datetime,
                "dishid": meal.dish
            }
        )
            .then(function (res) {
                var response = res;
                MealsService.List(userid, callback);
            })
            .catch(function (error) {
                console.log("error occured: ", error);
            });
    }
}